lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -207,4 +207,17 @@ void renameDirectory(Location source, Location target)
      */
     Set<Location> listDirectories(Location location)
             throws IOException;
+
+    /**
+     * Creates a temporary directory for the target path. The directory will be created
+     * using the (possibly absolute) prefix such that the directory can be renamed to
+     * the target path. The relative prefix will be used if required for the target path.
+     * <p>
+     * The temporary directory is not created for non-hierarchical file systems or for
+     * target paths that do not support renaming.
+     *
+     * @throws IllegalArgumentException If the target path is not valid for this file system.
```

Comment:
"or if the directory could not be created"

Reply:
That would be `IOException` which we don't document for the other methods.